frontend/util.lua

```diff
@@ -185,52 +185,18 @@ function util.secondsToHClock(seconds, withoutSeconds, hmsFormat)
             end
         end
     else
-        local round = withoutSeconds and require("optmath").round or passthrough
-        local hours = string.format("%d", seconds / 3600)
-        local mins = string.format("%02d", round(seconds % 3600 / 60))
+        local time_string = util.secondsToClock(seconds, withoutSeconds)
         if withoutSeconds then
-            if mins == "60" then
-                mins = string.format("%02d", 0)
-                hours = string.format("%d", hours + 1)
-            end
-            if hours == "0" then
-                -- We can optimize out the % 3600 since the branch ensures we're < than 3600
-                mins = string.format("%d", round(seconds / 60))
-                if hmsFormat then
-                    return T(_("%1m"), mins)
-                else
-                    return mins .. "'"
-                end
-            end
-            -- @translators This is the 'h' for hour, like in 1h30. This is a duration.
-            return T(_("%1h%2"), hours, mins)
+            time_string = time_string .. ":"
+        end
+        if hmsFormat then
+            time_string = time_string:gsub(":", "h", 1)
+            time_string = time_string:gsub(":", "m", 1)
+            return time_string .. (withoutSeconds and "" or "s")
```

Comment:
`return withoutSeconds and time_string or (time_string .. "s")`?

(Saves a useless concant w/ withoutSeconds ;)).